### ISSUE TITLE
[discussion] Investigating attention drift in ψ0 Qwen: reduced instruction grounding and arm/hand dominance

### ISSUE THREAD

**Issue by daniel-gieseler** [physical-superintelligence-lab/Psi0#48]
Just for context I am from the same team as the reproduction in issue #39.
We are interested in understanding the generalization of these models. So we built a small viewer to compare attention between vanilla Qwen and the ψ0 Qwen. Two things worth sharing kept showing up:
1. Instruction tokens lose per-word grounding.
2. Arms and hands take over the attention.

We realize that interpretability results are always tricky to reason about, but we’d really appreciate your perspective.

# Method

Following [this paper](https://arxiv.org/abs/2004.10102) we visualize each input token's contribution to the residual stream rather than raw attention weights — it's theoretically cleaner and in practice gives much sharper maps. We average over the last 4 layers.

# Reading the images

Each figure is a side-by-side: **left** is the ψ0 Qwen, **right** is vanilla Qwen3-VL-2B-Instruct. Same frame, same instruction ("pick up the sneaker and place it inside the box."). Overlays use two tiers — **red** is the top 10% of contribution, **orange** the top 30%.

# 1 — Instruction tokens lose per-word grounding

In vanilla Qwen, "box" lands on the box and "sneaker" on the sneaker. In the ψ0 Qwen that specificity is gone — the same tokens attend diffusely and look somewhat similar to each other.

("sneakers" selected - two tokens)
<img width="679" height="280" alt="Image" src="https://github.com/user-attachments/assets/198dad3c-9b43-4e91-b627-6078de3922d4" />

("box" selected)
<img width="684" height="280" alt="Image" src="https://github.com/user-attachments/assets/5de66648-bcdc-4bb4-ba15-577c4f990a88" />

**Thoughts.** One framing we've been playing with: since the loss only lands on the action tokens, maybe the language tokens just don't have direct pressure to keep their referents stable, so some drift is expected. But we're not sure how far it actually goes — has the instruction effectively collapsed into a single task code, or is there still compositional structure in there? We'd love your take, especially because compositionality should be important for generalizing instructions.

# 2 — Arms and hands dominate the attention

In vanilla Qwen, attention spreads over the task-relevant objects. In the ψ0 Qwen, whenever the arms are in frame, most of the attention goes to them.

(all instruction tokens selected)
<img width="684" height="279" alt="Image" src="https://github.com/user-attachments/assets/201f44a5-29d4-408c-aaeb-4037b7d5b738" />

**Thoughts.** Our thoughts here go a bit beyond this particular observation, into the relationship between the VLM and the action expert more broadly. The pretraining objective (predicting a task-relevant 100ms delta in pose) may push the VLM to focus on the current pose and produce features that look like short-horizon Cartesian motion cues — maybe even stripped of much task-level understanding by that point. Curious for any hunches you have here, especially since the action expert is planning over a much longer window than the 100ms the VLM saw.

**Comment by songlin** [comment]
Hi Daniel, thanks for sharing such interesting observations. To be honest, we are not surprised that the attention is dominated by the robotic arms and hands. Actually, that is the purpose of our pretraining: to learn a better visual representation for robotics. During pretraining, the Qwen3VL VLM model is fine-tuned to predict one-step future action at the centimeter level. The VLM must focus on the current robot arm and hand joints. And through your beautiful visualization, I think the pre-training has learned meaningful visual representation from ***human video*** and transferred it to the robots. The downside of such pretraining may shift the VLM's attention away from language grounding. We leave this as future exploration. 

Best,
Songlin
